### ISSUE TITLE
Rename `ingester` references in dskit/ring

### ISSUE THREAD

**Issue by DylanGuedes** [grafana/dskit#98]
Not a biiig concern, but I noticed that `dskit/ring` mentions ingester everywhere. I think changing these mentions to something more generic is beneficial because at first glance isn't clear if that code was applied for all rings or just for the ingester.

**Comment by pracucci** [comment]
I agree. We've already done part of this renaming long time ago and the generic name we picked is "instance". When continuing this work we need to pay attention to not introduce breaking changes (eg. protobuf serialization, metric names, ...).

**Comment by Sheikh-Abubaker** [comment]
@pracucci, @DylanGuedes I just went through `dskit/ring` and grepped ingester there are several lines of code where ingester is used, so do I need to rename each one of them to "instance" ?
